test/Microsoft.AspNetCore.Blazor.Test/TreeParameterStateTest.cs

```diff
@@ -0,0 +1,419 @@
+            var states = CreateAncestry(
+                CreateProvider((object)null),
+                new ComponentWithGenericTreeParam<TreeValue1>());
+
+            // Act
+            var result = TreeParameterState.FindTreeParameters(states.Last());
+
+            // Assert
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void FindTreeParameters_ProviderSpecifiesNameButConsumerDoesNot_ReturnsNull()
+        {
+            // Arrange
+            var states = CreateAncestry(
+                CreateProvider(new TreeValue1(), "MatchOnName"),
+                new ComponentWithTreeParams());
+
+            // Act
+            var result = TreeParameterState.FindTreeParameters(states.Last());
+
+            // Assert
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void FindTreeParameters_ConsumerSpecifiesNameButProviderDoesNot_ReturnsNull()
+        {
+            // Arrange
+            var states = CreateAncestry(
+                CreateProvider(new TreeValue1()),
+                new ComponentWithNamedTreeParam());
+
+            // Act
+            var result = TreeParameterState.FindTreeParameters(states.Last());
+
+            // Assert
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void FindTreeParameters_MismatchingNameButMatchingType_ReturnsNull()
+        {
+            // Arrange
+            var states = CreateAncestry(
+                CreateProvider(new TreeValue1(), "MismatchName"),
+                new ComponentWithNamedTreeParam());
+
+            // Act
+            var result = TreeParameterState.FindTreeParameters(states.Last());
+
+            // Assert
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void FindTreeParameters_MatchingNameButMismatchingType_ReturnsNull()
+        {
+            // Arrange
+            var states = CreateAncestry(
+                CreateProvider(new TreeValue2(), "MatchOnName"),
+                new ComponentWithNamedTreeParam());
+
+            // Act
+            var result = TreeParameterState.FindTreeParameters(states.Last());
+
+            // Assert
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void FindTreeParameters_MatchingNameAndType_ReturnsMatches()
+        {
+            // Arrange
+            var states = CreateAncestry(
+                CreateProvider(new TreeValue1(), "MatchOnName"),
+                new ComponentWithNamedTreeParam());
+
+            // Act
+            var result = TreeParameterState.FindTreeParameters(states.Last());
+
+            // Assert
+            Assert.Collection(result, match => {
+                Assert.Equal(nameof(ComponentWithNamedTreeParam.SomeLocalName), match.LocalName);
+                Assert.Same(states[0].Component, match.FromProvider);
+            });
+        }
+
+        [Fact]
+        public void FindTreeParameters_MultipleMatchingAncestors_ReturnsClosestMatches()
```

Comment:
is it worth having the same test, but where a `null` is the resulting value?

Reply:
OK, done.